PurchasesCoreSwiftTests/Mocks/MockProductsManager.swift

```diff
@@ -22,6 +22,18 @@ class MockProductsManager: ProductsManager {
         invokedProductsParametersList.append(identifiers)
         if let result = stubbedProductsCompletionResult {
             completion(result)
+        } else {
+            let products: [SKProduct] = identifiers.map { (identifier) -> MockSKProduct in
+                let p = MockSKProduct(mockProductIdentifier: identifier)
+                p.mockSubscriptionGroupIdentifier = "1234567"
+                if #available(iOS 12.2, *) {
+                    let mockDiscount = MockDiscount()
+                    mockDiscount.mockIdentifier = "discount_id"
+                    p.mockDiscount = mockDiscount
+                }
+                return p
+            }
+            completion(Set(products))
```

Comment:
the mock would previously not call completion at all if a stub had not been set, this seems more useful. It's taken from `MockRequestFetcher`

Reply:
also, having this means we can reuse the old testing code

Reply:
I'm a fan of adding more checks to our tests. Thanks for the update